eclair-core/src/main/scala/fr/acinq/eclair/Setup.scala

```diff
@@ -54,12 +54,22 @@ class Setup(datadir: File, overrideDefaults: Config = ConfigFactory.empty(), act
   implicit val formats = org.json4s.DefaultFormats
   implicit val ec = ExecutionContext.Implicits.global
 
+  /**
+    *
+    * @return the genesis hash of the blockchain we're on, in Big Endian format i.e. as computed by the hash
+    *         function (byte 0 first, byte 1 next and so on...). Bitcoin reverses block hashes and tx hashes and
+    *         these reversed hashes are often referred to as 'hashes' in the docs which adds to the confusion.
+    *         Theses hashes typically end with 0s, but are almost always reversed when displayed.
+    *         For example, the genesis hash for the testnet chain is 43497fd7f826957108f4a30fd9cec3aeba79972084e90ead01ea330900000000
+    */
+  def genesisHash: Future[BinaryData] = bitcoinClient.client.invoke("getblockhash", 0).map(json => BinaryData(json.extract[String]).reverse)
+
   val future = for {
     json <- bitcoinClient.client.invoke("getblockchaininfo")
     chain = (json \ "chain").extract[String]
     blockCount = (json \ "blocks").extract[Long]
     progress = (json \ "verificationprogress").extract[Double]
-    chainHash <- bitcoinClient.client.invoke("getblockhash", 0).map(_.extract[String])
+    chainHash <- genesisHash
```

Comment:
I wonder if we should not do this call but rather:

```scala
val chainHash = chain match {
    case "regtest" => Block.RegtestGenesisBlock.hash
    case "test" => Block.TestnetGenesisBlock.hash
    case _ => throw new RuntimeException("only regtest and testnet are supported for now")
  }
```

This is more consistent with the rest of this PR, and would simplify spv integration.